### PR TITLE
QueryIterator tracks yield counts in addition to next and seek counts.

### PR DESCRIPTION
**Pull request by apmoriarty**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
     
     protected long seek = 0;
     
+    protected long yield = 0;
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java]: This should probably be a boolean.  Either we yielded or we didn't

File: warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpanCollector.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 public class QuerySpanCollector {
     private AtomicLong seekCount = new AtomicLong();
     private AtomicLong nextCount = new AtomicLong();
+    private AtomicLong yieldCount = new AtomicLong();
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpanCollector.java]: AtomicBoolean?

File: warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java]: Updated

File: warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpanCollector.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpanCollector.java]: Changed to AtomicBoolean

File: warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java [hunk 3]
```diff
@@ -94,10 +96,22 @@ public long getSeekCount() {
...
+    public boolean getYield() {
+        if (yield) {
+            return true;
+        }
+        for (QuerySpan subSpan : sources) {
+            if (subSpan.getYield()) {
+                return true;
+            }
+        }
+        return true;
```
[the remark this answers is not in the document]
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/iterator/profile/QuerySpan.java]: Yes it should be. I'll update this shortly

**Comment by ivakegg** [comment]
OK, so what you have looks good, but you are missing the part where the metrics are combined on the webserver side and subsequently put into the query metrics.  See DocumentTransformerSupport.extractMetrics() and writeQueryMetrics(metric).

File: warehouse/query-core/src/main/java/datawave/query/attributes/TimingMetadata.java [hunk 4]
```diff
@@ -39,32 +40,34 @@ public long getSeekCount() {
...
     
-    public long getNextCount() {
-        Numeric numericValue = (Numeric) get(NEXT_COUNT);
+    public long getYieldCount() {
+        Numeric numericValue = (Numeric) get(YIELD_COUNT);
         if (numericValue != null) {
             return ((Number) numericValue.getData()).longValue();
         } else {
-            return 0;
+            return 0L;
```
Comment (alerman) [inline warehouse/query-core/src/main/java/datawave/query/attributes/TimingMetadata.java]: why 0L here, and not change the above to also use L

**Pull request merged**